[FX] Add wrap() docstring to docs and add decorator example

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50562 [WIP][FX] new sections in docs
* **#50555 [FX] Add wrap() docstring to docs and add decorator example**

![image](https://user-images.githubusercontent.com/4685384/104665715-edca5a00-5686-11eb-8439-45b48127b038.png)



Differential Revision: [D25917564](https://our.internmc.facebook.com/intern/diff/D25917564)

Commit:
[FX] Add wrap() docstring to docs and add decorator example

[ghstack-poisoned]

Commit:
Update on "[FX] Add wrap() docstring to docs and add decorator example"


![image](https://user-images.githubusercontent.com/4685384/104665715-edca5a00-5686-11eb-8439-45b48127b038.png)



Differential Revision: [D25917564](https://our.internmc.facebook.com/intern/diff/D25917564)

[ghstack-poisoned]

suo commented on torch/fx/symbolic_trace.py:
```diff
@@ -392,8 +392,8 @@ def _unpatch_wrapped_functions(orig_fns : List[PatchedFn]):
     This function can be called at global scope in a module to cause
-    references to the global function secified by `fn_name` to use
-    them in FX.
+    references to the global function specified by `fn_or_name` to use
```
I think this line is sort of unclear. In particular, I'm not sure what "global" means or "use them in FX" means.

Maybe something like:
This function can be called at module-level scope to register `fn_or_name` as a "leaf function". A "leaf function" will be preserved as a CallFunction node in the FX trace instead of being traced through

Commit:
Update on "[FX] Add wrap() docstring to docs and add decorator example"


![image](https://user-images.githubusercontent.com/4685384/104665715-edca5a00-5686-11eb-8439-45b48127b038.png)



Differential Revision: [D25917564](https://our.internmc.facebook.com/intern/diff/D25917564)

[ghstack-poisoned]

Chillee reviewed (approved):
LGTM
